### PR TITLE
[Feature][Connector-V2][Github] Adding Github Source Connector

### PR DESCRIPTION
**Pull request by Al-assad**
## Purpose of this pull request

Adding Github API data source connector based on http-connector.
Related issue: https://github.com/apache/incubator-seatunnel/issues/3018

## Check list

* [x] Code changed are covered with tests.
* [ ] If any new Jar binary package adding in your PR, please add License Notice according
  [New License Guide](https://github.com/apache/incubator-seatunnel/blob/dev/docs/en/contribution/new-license.md)
* [x] If necessary, please update the documentation to describe the new feature. https://github.com/apache/incubator-seatunnel/tree/dev/docs
* [x] If you are contributing the connector code, please check that the following files are updated:
  1. Update change log that in connector document. For more details you can refer to [connector-v2](https://github.com/apache/incubator-seatunnel/tree/dev/docs/en/connector-v2)
  2. Update [plugin-mapping.properties](https://github.com/apache/incubator-seatunnel/blob/dev/plugin-mapping.properties) and add new connector information in it
  3. Update the pom file of [seatunnel-dist](https://github.com/apache/incubator-seatunnel/blob/dev/seatunnel-dist/pom.xml)
* [ ] Update the [`release-note`](https://github.com/apache/incubator-seatunnel/blob/dev/release-note.md).

### CODE REVIEW

**Comment by Al-assad** [comment]
@CalvinKirs @TyrantLucifer @EricJoy2048 @ashulin PTAL.

File: seatunnel-e2e/seatunnel-connector-v2-e2e/connector-http-e2e/src/test/java/org/apache/seatunnel/e2e/connector/http/HttpGithubIT.java [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+public class HttpGithubIT extends HttpIT {
+
+    @Override
+    public String getITTestConf() {
+        return "/github_json_to_assert.conf";
+    }
+
+    @Override
+    public String getMockServerConfig() {
+        return "/mockserver-gitlab-config.json";
```
Comment (liugddx) [inline seatunnel-e2e/seatunnel-connector-v2-e2e/connector-http-e2e/src/test/java/org/apache/seatunnel/e2e/connector/http/HttpGithubIT.java]:
```suggestion
        return "/mockserver-github-config.json";
```

File: seatunnel-connectors-v2/connector-http/connector-http-github/pom.xml [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+        <dependency>
+            <groupId>org.apache.seatunnel</groupId>
+            <artifactId>connector-http-base</artifactId>
+            <version>${project.version}</version>
+        </dependency>
+        <dependency>
+            <groupId>org.apache.seatunnel</groupId>
+            <artifactId>seatunnel-api</artifactId>
+            <scope>compile</scope>
+        </dependency>
```
Comment (TyrantLucifer) [inline seatunnel-connectors-v2/connector-http/connector-http-github/pom.xml]: Remove useless dependency.

File: seatunnel-connectors-v2/connector-http/connector-http-github/src/main/java/org/apache/seatunnel/connectors/seatunnel/github/GithubPlugin.java [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.seatunnel.connectors.seatunnel.github;
+
+public class GithubPlugin {
+
+    public static final String PLUGIN_NAME = "Github";
```
Comment (TyrantLucifer) [inline seatunnel-connectors-v2/connector-http/connector-http-github/src/main/java/org/apache/seatunnel/connectors/seatunnel/github/GithubPlugin.java]: Remove it to `GithubSource`

File: seatunnel-connectors-v2/connector-http/connector-http-github/src/main/java/org/apache/seatunnel/connectors/seatunnel/github/source/GithubSource.java [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+    public String getPluginName() {
+        return GithubPlugin.PLUGIN_NAME;
+    }
+
+    @Override
+    public Boundedness getBoundedness() throws UnsupportedOperationException {
+        if (JobMode.BATCH.equals(jobContext.getJobMode())) {
+            return Boundedness.BOUNDED;
+        }
+        throw new UnsupportedOperationException(
```
Comment (TyrantLucifer) [inline seatunnel-connectors-v2/connector-http/connector-http-github/src/main/java/org/apache/seatunnel/connectors/seatunnel/github/source/GithubSource.java]: Unified exception, you can refer to this guide: https://github.com/apache/incubator-seatunnel/issues/2930

File: seatunnel-connectors-v2/connector-http/connector-http-github/src/main/java/org/apache/seatunnel/connectors/seatunnel/github/source/GithubSource.java [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+                "Github source connector not support unbounded operation");
+    }
+
+    @Override
+    public void prepare(Config pluginConfig) throws PrepareFailException {
+        CheckResult result =
+                CheckConfigUtil.checkAllExists(pluginConfig, GithubSourceConfig.URL.key());
+        if (!result.isSuccess()) {
+            throw new PrepareFailException(getPluginName(), PluginType.SOURCE, result.getMsg());
```
Comment (TyrantLucifer) [inline seatunnel-connectors-v2/connector-http/connector-http-github/src/main/java/org/apache/seatunnel/connectors/seatunnel/github/source/GithubSource.java]: Unified exception.

**Comment by Al-assad** [comment]
@liugddx @TyrantLucifer Thanks for your suggestions, please take a look agian.

**Review by TyrantLucifer — COMMENTED** [review]
Update https://github.com/apache/incubator-seatunnel/blob/dev/seatunnel-dist/pom.xml

**Comment by TyrantLucifer** [comment]
> Update https://github.com/apache/incubator-seatunnel/blob/dev/seatunnel-dist/pom.xml

Sorry my fault, you also should update release-note https://github.com/apache/incubator-seatunnel/blob/dev/release-note.md too

**Comment by TyrantLucifer** [comment]
> > Update https://github.com/apache/incubator-seatunnel/blob/dev/seatunnel-dist/pom.xml
> 
> Sorry my fault, you also should update release-note https://github.com/apache/incubator-seatunnel/blob/dev/release-note.md too

image

**Comment by Al-assad** [comment]
> Update https://github.com/apache/incubator-seatunnel/blob/dev/seatunnel-dist/pom.xml

Resolved 😃.

**Comment by liugddx** [comment]
> > Update https://github.com/apache/incubator-seatunnel/blob/dev/seatunnel-dist/pom.xml
> 
> Resolved 😃.

You should update `release-note.md` for this pr.

**Review by TyrantLucifer — DISMISSED** [review]
Overall LGTM, let's waiting CI. cc @liugddx

**Review by liugddx — DISMISSED** [review]
LGTM

**Review by liugddx — APPROVED** [review]

**Comment by Al-assad** [comment]
> 


> 
I just submitted an update to the release-note and the approval status will be dimissed again in the github mechanism, please re-add your approval, thanks.

**Review by TyrantLucifer — APPROVED** [review]

**Pull request merged**